File: algo-anagrams-i/js/characterMatch.js
exports.isCharacterMatch = function(string1, string2) {
    arr1 = string1.toLowerCase().replace(/\s/g, "").split("").sort();
    arr2 = string2.toLowerCase().replace(/\s/g, "").split("").sort();
    let longest = 0;
    if (arr1.length > arr2.length) {
        longest = arr1.length;
    } else {
        longest = arr2.length;
    }
    for (let i = 0; i < longest; i++) {
        if (arr1[i] !== arr2[i]) {
            return false;
        }
    }
    return true;
};

exports.anagramsFor = function(word, listOfWords) {
    output = [];
    for (let wordTwo of listOfWords) {
        if (this.isCharacterMatch(word, wordTwo)) {
            output.push(wordTwo);
        }
    }
    return output;
};

File: algo-anagrams-i/python/character_match.py
# Don't forget to run the tests (and create some of your own)

# Part 1
def is_character_match(string1, string2):
    updated_string1 = string1.lower().replace(" ", "")
    updated_string2 = string2.lower().replace(" ", "")
    list1 = sorted([*updated_string1])
    list2 = sorted([*updated_string2])
    longest = 0
    if (len(list1) > len(list2)):
        longest = len(list1)
    else:
        longest = len(list2)
    for num in range(longest):
        if list1[num] != list2[num]:
            return False
    return True


# Part 2
def anagrams_for(word, list_of_words):
    output = []
    for wordTwo in list_of_words:
        if is_character_match(word, wordTwo):
            output.append(wordTwo)
    return output
